neurom/features/neurite.py

```diff
@@ -65,18 +65,23 @@
 L = logging.getLogger(__name__)
 
 
+def _return_1(*args, **kwargs):
+    # pylint: disable=unused-argument
+    return 1
+
+
 def _map_sections(fun, neurite, iterator_type=Section.ipreorder, section_type=NeuriteType.all):
     """Map `fun` to all the sections."""
     check_type = is_type(section_type)
 
     def homogeneous_filter(section):
-        return check_type(section) and Section.is_homogeneous_point(section)
+        return check_type(section) and all(check_type(c) for c in section.children)
```

Comment:
`Section.is_homogeneous_point` is not sufficient here?

Reply:
Not when the user passes a composite type because the current section and its children might not have the same type since sections do not have a composite type.
Another possible implementation could be to add an optional `composite_type` argument to `Section.is_homogeneous_point`. So when it is not passed it just compares the types of the children to the type of the current section and when it is passed it compares to this passed type (and I guess we should also check that the current section is contained in this composite type, otherwise it could lead to weird results). Does this implementation makes sense to you?